From 5bd37dfe58e6a1230af01cdf0524a0ddea36b735 Mon Sep 17 00:00:00 2001
From: Luke Morrison <lfmorrison@gmail.com>
Date: Tue, 16 Jan 2018 10:56:16 -0400
Subject: [PATCH 071/249] device_build.md: Handle split userspace/CPU
 architecture

The build instructions have some conditional content that is only
applicable to devices with the x86 architecture. Update the build
instructions to parse devices with a split CPU/userspace architecture
property properly, while remaining compatible with devices that still
only have the single overall architecture property.

Change-Id: I574360d2ee7d0e5e409cd8df49642987e32984d2
---
 _includes/templates/device_build.md | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/_includes/templates/device_build.md b/_includes/templates/device_build.md
index 7964351..71ba4e6 100644
--- a/_includes/templates/device_build.md
+++ b/_includes/templates/device_build.md
@@ -67,12 +67,20 @@ Several packages are needed to build LineageOS. You can install these using your
 (usually originating from the Internet) on your computer. With Ubuntu, you can use the Ubuntu Software Center. Even better, you may also use the `apt-get install`
 command directly in the Terminal." %}
 
+{%- capture cpu_architecture %}
+{%- if device.architecture.cpu -%}
+   {{ device.architecture.cpu }}
+{%- else -%}
+   {{ device.architecture }}
+{%- endif -%}
+{%- endcapture -%}
+
 To build LineageOS, you'll need:
 
 * `bc bison build-essential ccache curl flex g++-multilib gcc-multilib git gnupg gperf imagemagick
    lib32ncurses5-dev lib32readline-dev lib32z1-dev libesd0-dev liblz4-tool libncurses5-dev
    libsdl1.2-dev libssl-dev libwxgtk3.0-dev libxml2 libxml2-utils lzop pngcrush rsync
-   schedtool squashfs-tools xsltproc {% if device.architecture contains 'x86' %}yasm {% endif %}
+   schedtool squashfs-tools xsltproc {% if cpu_architecture contains 'x86' %}yasm {% endif %}
    zip zlib1g-dev`
 
 {% if device.versions contains 13.0 %}
-- 
2.17.1

